src/Autofac.Extras.Moq/AutoMock.cs

```diff
@@ -42,18 +42,19 @@ public class AutoMock : IDisposable
 
         private readonly List<Type> _createdServiceTypes = new List<Type>();
 
-        private AutoMock(MockBehavior behavior)
-            : this(new MockRepository(behavior))
+        private AutoMock(MockBehavior behavior, Action<ContainerBuilder> beforeBuild)
+            : this(new MockRepository(behavior), beforeBuild)
         {
         }
 
-        private AutoMock(MockRepository repository)
+        private AutoMock(MockRepository repository, Action<ContainerBuilder> beforeBuild)
         {
             this.MockRepository = repository;
             var builder = new ContainerBuilder();
             builder.RegisterInstance(this.MockRepository);
             builder.RegisterSource(new AnyConcreteTypeNotAlreadyRegisteredSource());
             builder.RegisterSource(new MoqRegistrationHandler(_createdServiceTypes));
+            beforeBuild?.Invoke(builder);
```

Comment:
After I take this pull, I will probably put this _after_ the `RegisterInstance` but _before_ the `RegisterSource` calls to help alleviate issues like autofac/Autofac#341.